include/lbann/layers/io/input/generic_input_layer.hpp

```diff
@@ -558,16 +564,21 @@ class generic_input_layer : public io_layer {
    * Return the sample indices fetched in the current mini-batch.
    */
   El::Matrix<El::Int>* get_sample_indices_per_mb() override {
-    execution_mode mode = this->m_model->get_execution_mode();
+    execution_mode mode = this->m_model->get_execution_context().get_execution_mode();
     generic_io_buffer* io_buffer = m_io_buffers[get_active_buffer_idx(mode) % m_io_buffers.size()];
-    return io_buffer->get_sample_indices_fetched_per_mb(this->m_model->get_execution_mode());
+    return io_buffer->get_sample_indices_fetched_per_mb(this->m_model->get_execution_context().get_execution_mode());
   }
 
   /**
    * Get the dimensions of the underlying data.
    */
   const std::vector<int> get_data_dims(int child_index = 0) const override {
-    const generic_data_reader *dr = get_data_reader();
+    // Check the training and testing execution modes for data dimensions
+    const generic_data_reader *dr = get_data_reader(execution_mode::training);
+    if(dr != nullptr) {
+      dr = get_data_reader(execution_mode::testing);
+    }
+    if(dr == nullptr) { LBANN_ERROR("unable to call get_data_dims -- no valid execution mode"); }
```

Comment:
```suggestion
    else { LBANN_ERROR("unable to call get_data_dims -- no valid execution mode"); }
```

Reply:
I could imagine this is fine as is, if we want to check if `get_data_reader` returns `nullptr`. Is this the intention?

Reply:
Oh I missed that `dr` gets reset in the `if` block... My suggestion should probably be ignored.